File: 20251111_Contenido_info.js
/*
Video: https://www.youtube.com/watch?v=NUV8ltTm4eA

Cómo producir la cantidad de vocales y consonantes de una frase con JavaScript:

Para retornar una matriz que contiene la cantidad de vocales y consonantes de una cadena de caracteres, podemos crear una función personalizada que acepta una cadena como parámetro. Creamos una variable para almacenar el total en una matriz. Creamos dos variables adicionales: una para obtener la cantidad de vocales y la otra para los consonantes. Agregamos esas dos últimas variables a la variable con la matriz y la retornamos. Después simplemente lo mandamos a la consola.
*/

function cuenta(str) {
  let total = [];
  const vocales = str.match(/[aeiou]/gi);
  const consonantes = str.match(/[bcdfghjklmnpqrstvwxyz]/gi);
  vocales ? total.push(vocales.length) : 0;
  consonantes ? total.push(consonantes.length) : 0;
  
  return total
}

console.log(cuenta("Hola mundo"));

File: 20251111_Contenido_info.py
"""
Video: https://www.youtube.com/watch?v=6cZwKKVG-Vk

Cómo producir la cantidad de vocales y consonantes de una frase con Python:

Para retornar una lista que contiene la cantidad de vocales y consonantes de una cadena de caracteres, podemos crear una función personalizada que acepta una cadena como parámetro. Creamos una variable para almacenar el total en una lista. Creamos dos variables adicionales: una para obtener la cantidad de vocales y la otra para los consonantes. Agregamos esas dos ultimas variables a la variable con la lista y la retornamos. Después simplemente imprimimos.
"""

def cuenta(s):
    total = []
    vocales = len([c for c in s.lower() if c in 'aeiou'])
    consonantes = len([c for c in s.lower() if c in 'bcdfghjklmnpqrstvwxyz'])
    total.append(vocales)
    total.append(consonantes)
    return total

print(cuenta("Hola mundo"))
